### PR TITLE
FixXsdDataContractExporterTestAndFragmentTests

### LINKED ISSUE #12772: XsdDataContractExporterTest Fail
Throw null reference exception when call the following code
 XsdDataContractExporter exporter = new XsdDataContractExporter();
exporter.CanExport(typeof(Employee));

This is because DataContractSet returns null (XsdDataContractExporter.cs). The code was commented.
DataContractSet DataContractSet
        {
            get
            {
                //if (dataContractSet == null)
                //{
                //    dataContractSet = new DataContractSet((Options == null) ? null : Options.GetSurrogate());
                //}
                return _dataContractSet;
            }
        }

### LINKED ISSUE #13375: IFragmentCapableXmlDictionaryWriter not implemented by XmlBaseWriter
IFragmentCapableXmlDictionaryWriter is not implemented by any class.  In desktop, it is implemented by XmlBaseWriter.

Our code:
internal abstract class XmlBaseWriter : XmlDictionaryWriter
{...}

Code in desktop.
abstract class XmlBaseWriter : XmlDictionaryWriter, IFragmentCapableXmlDictionaryWriter
{...}

### PR DESCRIPTION
**Pull request by huanwu**
1.Throw PlatformNotSupportedException when to call DataContractSet since it always returns null.
2. Fix #12772.
3. Fix #13375

@shmao @mconnew @zhenlan

### CODE REVIEW

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XsdDataContractExporter.cs [hunk 1]
```diff
@@ -62,7 +62,8 @@ DataContractSet DataContractSet
                 //{
                 //    dataContractSet = new DataContractSet((Options == null) ? null : Options.GetSurrogate());
                 //}
-                return _dataContractSet;
+                //return _dataContractSet;
```
Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XsdDataContractExporter.cs]: Remove the code being commented out in this method.

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 2]
```diff
@@ -287,33 +287,19 @@ public static void IXmlTextReaderInitializerTest()
...
     {
         string rwTypeStr = "Text";
         ReaderWriterFactory.ReaderWriterType rwType = (ReaderWriterFactory.ReaderWriterType)
             Enum.Parse(typeof(ReaderWriterFactory.ReaderWriterType), rwTypeStr, true);
         Encoding encoding = Encoding.GetEncoding("utf-8");
-        int numberOfNestedFragments = 1;
         MemoryStream ms1 = new MemoryStream();
         MemoryStream ms2 = new MemoryStream();
         XmlDictionaryWriter writer1 = (XmlDictionaryWriter)ReaderWriterFactory.CreateXmlWriter(rwType, ms1, encoding);
         XmlDictionaryWriter writer2 = (XmlDictionaryWriter)ReaderWriterFactory.CreateXmlWriter(rwType, ms2, encoding);
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Do we still need both of the above lines? Do we need any of them?

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 3]
```diff
@@ -287,33 +287,19 @@ public static void IXmlTextReaderInitializerTest()
...
-        writer1.WriteEndElement(); writer2.WriteEndElement();
-        writer1.WriteEndDocument(); writer2.WriteEndDocument();
-        writer1.Flush();
-        writer2.Flush();
-
-        byte[] doc1 = ms1.ToArray();
-        byte[] doc2 = ms2.ToArray();
-        CompareArrays(doc1, 0, doc2, 0, doc1.Length);
+        Assert.False(FragmentHelper.CanFragment(writer1));
+        Assert.False(FragmentHelper.CanFragment(writer2));
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Do we need both of the above lines?

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 4]
```diff
@@ -414,15 +400,6 @@ private static string GetExpectString(byte[] bytes, int byteSize)
...
     }
-
-    private static void CompareArrays(byte[] array1, int offset1, byte[] array2, int offset2, int count)
-    {
-        for (int i = 0; i < count; i++)
-        {
-            Assert.Equal(array1[i + offset1], array2[i + offset2]);
-        }
-    }
-
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Nit: add an empty line between two methods.

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs (hunk 2 above)
Comment (huanwu) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: I will keep half of them so we have more API coverage.

Comment (huanwu) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: I will remove and keep half of them so we have more API coverage.

**Pull request merged**